designs/administrator/future-openshift/import-multiple-yaml/index.md

```diff
@@ -0,0 +1,38 @@
+---
+parent: Administrator
+version: 4.x
+---
+
+# Import YAML for multiple resources
+This design details how users can import multiple resources using the **Import YAML** page.
+
+## Design
+
+The **Import YAML** page will still be accessible from the masthead. There are three ways a user can create multiple resources on this page. The instructions on the page should be updated to clarify how users can import YAML: *Drag and drop YAML or JSON files into the editor, or manually enter files and use `- - -` to separate each definition.*
+
+1) Users can manually enter multiple definitions in the editor, using `- - -` to delineate them.
+2) – Users can drag and drop a single YAML with multiple resources defined and separated with `- - -`.
+
+Both methods will show the following in the editor:
+
+![1](img/manual-entry.png)
+
+3) Users can drag and drop multiple files into the YAML area at once.
+
+![2](img/draganddrop.png)
+
+After the user clicks **Create**, the console will do a dry run of resource creation to find potential errors before beginning to actually create the resources. If errors are found, the user will be notified by an inline alert on the **Import YAML** page beneath the editor. If the dry run is completed without any issues, the resources will begin to be created and the user will be redirected to the **Importing resources** page.
+
+![3](img/importing.png)
+
+The **Importing resources** page will show a table with all the resources the user is creating. The table will display the resource name, namespace, and creation status. When the resources are successfully created, they should turn into a blue link that redirects users to the resource's details page.
```

Comment:
I'm assuming the creation of each resource is so quick that the user probably wouldn't navigate away using the links before it's complete?

Otherwise if they do navigate away, I assume the process continues in the background?

Reply:
Yep, I believe that creation should be nearly instantaneous, so definitely not a ton of time for users to navigate away. If they do, the process should continue in the background. Another good case for future toasts :)

Reply:
I'll add this to the documentation